bin/deliver_reviewed_data.py

```diff
@@ -0,0 +1,244 @@
+__author__ = 'tcezard'
+import sys
+import os
+import argparse
+import logging
+import datetime
+from collections import defaultdict
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from analysis_driver.config import default as cfg, logging_default as log_cfg
+from analysis_driver.app_logging import AppLogger
+from analysis_driver import executor, clarity, rest_communication
+from analysis_driver.exceptions import AnalysisDriverError
+
+
+
+hs_list_files = [
+    '{ext_sample_id}.g.vcf.gz',
+    '{ext_sample_id}.g.vcf.gz.tbi',
+    '{ext_sample_id}.vcf.gz',
+    '{ext_sample_id}.vcf.gz.tbi',
+    '{ext_sample_id}.bam',
+    '{ext_sample_id}.bam.bai',
+    '{ext_sample_id}_R1.fastq.gz',
+    '{ext_sample_id}_R2.fastq.gz',
+    '{ext_sample_id}_R1_fastqc.html',
+    '{ext_sample_id}_R2_fastqc.html',
+]
+
+other_list_files = [
+    '{ext_sample_id}_R1.fastq.gz',
+    '{ext_sample_id}_R2.fastq.gz',
+    '{ext_sample_id}_R1_fastqc.html',
+    '{ext_sample_id}_R2_fastqc.html',
+]
+
+
+class DataDelivery(AppLogger):
+
+    def __init__(self, dry_run):
+        self.all_commands = []
+        self.dry_run = dry_run
+        self.all_samples_values = []
+        self.sample2species = {}
+
+    def get_deliverable_projects_samples(self, project_id=None, sample_id=None):
+        project_to_samples = defaultdict(list)
+        #Get sample that have been review from REST API but not marked as delivered
+        where_clause={"useable":"yes"}
+        if project_id:
+            where_clause["project_id"] = project_id
+        if sample_id:
+            where_clause["sample_id"] = sample_id
+        # These samples are useable but could have been delivered already so need to check
+        samples = rest_communication.get_documents(
+                'samples',
+                depaginate=True,
+                embedded={"analysis_driver_procs":1, "run_elements":1},
+                where=where_clause
+        )
+        for sample in samples:
+            processes = sample.get('analysis_driver_procs',[{}])
+            processes.sort(key=lambda x: datetime.datetime.strptime(x.get('_created', '01_01_1970_00:00:00'), '%d_%m_%Y_%H:%M:%S'))
+            if not processes[-1].get('status','new') == 'finished':
+                raise AnalysisDriverError("Reviewed sample %s not marked as finished"%sample.get('sample_id'))
+            if sample.get('delivered','no') == 'no':
+                project_to_samples[sample.get('project_id')].append(sample.get('sample_id'))
+                self.all_samples_values.append(sample)
```

Comment:
If all samples have `delivered=no` as a default, we can put the `if` statement into the `where` filter